### ISSUE TITLE
Reorder error functions inputs y_true and y_pred

### ISSUE THREAD

**Issue by mratsim** [mratsim/Arraymancer#380]
All errors functions like `accuracy_score`, `relative_error` and `absolute_error` follow the following signature convention

```Nim
proc my_error[T](y_true, y: Tensor[T]): T =
  discard
```

This is the same convention as scikit-learn, see [accuracy_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html).

However this convention hurts composition via the method call syntax.
For example, inverting the arguments would allow.
```
my_tensor.foo.bar.baz.accuracy_score(true_labels)
```
While currently we are "forced" to use function call syntax:
```
accuracy_score(true_labels, my_tensor.foo.bar.baz)
```

The potential breakage is minimal as all basic error functions are commutative. And it will not break named parameters calls.

Complex loss functions like `softmax_cross_entropy`, `sigmoid_cross_entropy` or `mean_loss` already use the (input, target) ordering.
